### PR TITLE
Refactor inventory script

### PR DESCRIPTION
**Pull request by leki75**
Modifications:
- pep8 compatibility
- virtualenv support (shebang)
- handle unknown hosts
- --list or --host are mutually exclusive

### CODE REVIEW

File: openstack-installer/inventory/inventory.py [hunk 1]
```diff
@@ -6,53 +6,69 @@
...
                 hosts.append(host)
+
     return hosts
 
+
 def inventory(hostname):
 
-    with open(os.path.dirname(sys.argv[0])+"/inventory.yml", 'r') as f:
-        yml=yaml.safe_load(f)
+    with open(os.path.join(os.path.dirname(sys.argv[0]), "inventory.yml"), 'r') as f:  # nopep8
```
Comment (gyurco) [inline openstack-installer/inventory/inventory.py:37]: no solution without #nopep8?

File: openstack-installer/inventory/inventory.py [hunk 2]
```diff
@@ -6,53 +6,69 @@
...
             if data and hostname in data:
-                hostvars=data[hostname]
-                hostvars['ansible_ssh_host']=hostvars['ip']['mgmt']
+                hostvars = data[hostname]
+                hostvars['ansible_ssh_host'] = hostvars['ip']['mgmt']
                 print json.dumps(hostvars, indent=4)
-                return
+                break
+        else:
+            sys.stderr.write("ERROR: No host found with name '{0}'\n".format(hostname))  # nopep8
```
Comment (gyurco) [inline openstack-installer/inventory/inventory.py:48]: no solution without #nopep8?

File: openstack-installer/inventory/inventory.py [hunk 3]
```diff
@@ -6,53 +6,69 @@
...
-            inventory[group]=sorted(expand_group(yml,group,0))
+            inventory[group] = sorted(expand_group(yml, group, 0))
             if data:
                 for host, hostvars in data.iteritems():
                     if host != 'inherit':
-                        inventory['_meta']['hostvars'][host]=hostvars
-                        inventory['_meta']['hostvars'][host]['ansible_ssh_host']=hostvars['ip']['mgmt']
-
+                        inventory['_meta']['hostvars'][host] = hostvars
+                        inventory['_meta']['hostvars'][host]['ansible_ssh_host'] = hostvars['ip']['mgmt']  # nopep8
```
Comment (gyurco) [inline openstack-installer/inventory/inventory.py:58]: no solution without #nopep8?

File: openstack-installer/inventory/inventory.py [hunk 4]
```diff
@@ -6,53 +6,69 @@
...
 
-parser = argparse.ArgumentParser(description='Dynamic inventory for Openstack.')
-parser.add_argument('--list', help='list the hosts', action='store_true')
-parser.add_argument('--host', help='returnt the variables for the hosts')
-args = parser.parse_args()
 
-inventory(args.host)
+def main():
+
+    parser = argparse.ArgumentParser(description='Dynamic inventory for Openstack.')  # nopep8
```
Comment (gyurco) [inline openstack-installer/inventory/inventory.py:64]: no solution without #nopep8?

File: openstack-installer/inventory/inventory.py [hunk 5]
```diff
@@ -6,53 +6,69 @@
...
 
+
 def expand_group(yml, group, nesting_level):
 
-    nesting_level+=1
+    nesting_level += 1
     if nesting_level == MAX_NESTING_LEVEL:
-        raise(Exception, "Too many nesting level of groups, probably a loop?")
-    hosts=[]
+        sys.stderr.write("ERROR: Too many nesting level of groups, probably a loop?\n")  # nopep8
```
Comment (gyurco) [inline openstack-installer/inventory/inventory.py:17]: no solution without #nopep8?

**Comment by leki75** [comment]
It is possible to split long lines to 2 or more.

**Comment by leki75** [comment]
Pep8 compatibility is merged into master the other modifications are split into separate commits.

**Pull request closed without merging**